plugins/lsp/lspsaga.nix

```diff
@@ -54,7 +54,7 @@ in
       ui = {
         border = helpers.defaultNullOpts.mkBorder "single" "lspsaga" "";
 
-        devicon = helpers.defaultNullOpts.mkBool true "Whether to use nvim-web-devicons.";
+        devicon = helpers.defaultNullOpts.mkBool config.iconsEnabled "Whether to use nvim-web-devicons.";
```

Comment:
This doesn't do what you think it does.

This is a "plugin default" not a nix option default, so this will have no effect.

If we actually want to do this, it should be done using a `mkIf` + `mkDefault` config definition, however I feel uneasy about setting freeform settings options "automatically".

@traxys any thoughts?

Reply:
Isn't the plugin config defined as `cfg = config.plugins.lspsaga` in this scope on L10? `config` seems to be the correct parent scope. It seemed to all work when I tested it.

Reply:
In `defaultNullOpts`, the "default" argument is _not_ the option default.

Instead, it is the "plugin default" and is only used for generating documentation. The _actual_ default is always `null`.

The "plugin default" has no effect on the option's value in nix. Rather it exists as a way to let users know what the upstream plugin will do when the option is undefined (`null`).

Reply:
Ah, okay, it's specific to `defaultNullOpts`. I can just revert my change to this file. Are other other `mkOption`s okay?